File: packages/sdk-typescript/examples/quickstart.ts
import { OffchainClient, OrderSide, OrderType } from '@offchain-protocol/sdk';

async function main() {
  // Initialize the client
  const client = new OffchainClient('your-api-key', {
    testnet: true
  });

  try {
    // Connect to WebSocket for real-time updates
    await client.connect();
    console.log('Connected to Offchain Protocol');

    // Create a limit order
    const order = await client.orders.create({
      pair: 'BTC/USDT',
      side: OrderSide.BUY,
      type: OrderType.LIMIT,
      quantity: '0.1',
      price: '45000'
    });
    console.log('Order created:', order.id);

    // Get order book
    const orderBook = await client.orderBook.get('BTC/USDT');
    console.log('Best bid:', orderBook.bids[0]?.price);
    console.log('Best ask:', orderBook.asks[0]?.price);

    // Subscribe to order updates
    client.websocket.subscribeOrders();
    client.websocket.on('order:update', (updatedOrder) => {
      console.log('Order updated:', updatedOrder.id, updatedOrder.status);
    });

    // Stream trades
    client.websocket.subscribeTrades(['BTC/USDT']);
    client.websocket.on('trade', (trade) => {
      console.log(`Trade: ${trade.price} x ${trade.quantity}`);
    });

    // Keep the connection alive
    process.on('SIGINT', async () => {
      console.log('\nClosing connection...');
      await client.disconnect();
      process.exit(0);
    });

  } catch (error) {
    console.error('Error:', error);
    await client.disconnect();
  }
}

main().catch(console.error);

File: packages/sdk-python/examples/quickstart.py
import asyncio
from offchain_protocol import OffchainClient, CreateOrderRequest, OrderSide, OrderType

async def main():
    # Initialize the client
    async with OffchainClient('your-api-key', testnet=True) as client:
        print('Connected to Offchain Protocol')
        
        try:
            # Create a limit order
            order = await client.create_order(CreateOrderRequest(
                pair='BTC/USDT',
                side=OrderSide.BUY,
                type=OrderType.LIMIT,
                quantity='0.1',
                price='45000'
            ))
            print(f'Order created: {order.id}')
            
            # Get order book
            orderbook = await client.get_orderbook('BTC/USDT')
            if orderbook.bids:
                print(f'Best bid: {orderbook.bids[0].price}')
            if orderbook.asks:
                print(f'Best ask: {orderbook.asks[0].price}')
            
            # Get recent trades
            trades = await client.get_recent_trades('BTC/USDT', limit=10)
            for trade in trades[:5]:  # Show first 5
                print(f'Trade: {trade.price} x {trade.quantity} at {trade.timestamp}')
            
            # Stream real-time trades
            print('\nStreaming trades (press Ctrl+C to stop)...')
            async for trade in client.stream_trades('BTC/USDT'):
                print(f'New trade: {trade.price} x {trade.quantity}')
                
        except KeyboardInterrupt:
            print('\nStopping...')
        except Exception as e:
            print(f'Error: {e}')

if __name__ == '__main__':
    asyncio.run(main())